java/arcs/core/storage/ReferenceModeStore.kt

```diff
@@ -319,7 +337,24 @@ class ReferenceModeStore private constructor(
                 if (pendingIds.isEmpty()) {
                     sendQueue.enqueue(::sender)
                 } else {
-                    sendQueue.enqueueBlocking(pendingIds, ::sender)
+                    try {
+                        withTimeout(BLOCKING_QUEUE_TIMEOUT_MILLIS) {
+                            sendQueue.enqueueBlocking(pendingIds, ::sender).await()
```

Comment:
I wonder if the timeout functionality could be built into the queue?

Reply:
I had thought about that.. it'd definitely be cleaner that way.  I think if we end up using the SendQueue outside of the ReferenceModeStore (for example, with Alice's work), then it would be worth the effort.